docs/types/reference-types.rst

```diff
@@ -28,6 +28,9 @@ annotation, the "data location", about where it is stored. There are three data
 ``memory``, ``storage`` and ``calldata``. Calldata is a non-modifiable,
 non-persistent area where function arguments are stored, and behaves mostly like memory.
 
+.. note::
+    ``transient storage`` is not yet supported as a data location for reference types.
```

Comment:
I think we should mention `transient` in some way in "Value Types" too.

Overall, it looks like there's currently only one example of a transient variable in the whole docs and it's in "Inline Assembly". I.e. we barely mention it at all and we pretty much never show it so it feels like it's never being introduced properly compared to other language features.